programs/uxd/src/instructions/mercurial/collect_profit_of_mercurial_vault_depository.rs

```diff
@@ -0,0 +1,288 @@
+
+    /// #9
+    #[account(mut)]
+    pub mercurial_vault_lp_mint: Box<Account<'info, Mint>>,
+
+    /// #10
+    /// Token account owned by the mercurial vault program. Hold the collateral deposited in the mercurial vault.
+    #[account(mut)]
+    pub mercurial_vault_collateral_token_safe: Box<Account<'info, TokenAccount>>,
+
+    /// #11
+    pub mercurial_vault_program: Program<'info, mercurial_vault::program::Vault>,
+
+    /// #12
+    pub system_program: Program<'info, System>,
+
+    /// #13
+    pub token_program: Program<'info, Token>,
+}
+
+pub fn handler(ctx: Context<CollectProfitOfMercurialVaultDepository>) -> Result<()> {
+    let depository = ctx.accounts.depository.load()?;
+    let mercurial_vault: Pubkey = depository.mercurial_vault;
+    let collateral_mint = depository.collateral_mint;
+    let depository_bump = depository.bump;
+    drop(depository);
+
+    let depository_signer_seed: &[&[&[u8]]] = &[&[
+        MERCURIAL_VAULT_DEPOSITORY_NAMESPACE,
+        mercurial_vault.as_ref(),
+        collateral_mint.as_ref(),
+        &[depository_bump],
+    ]];
+
+    let before_lp_token_vault_balance = ctx.accounts.depository_lp_token_vault.amount;
+    let before_collateral_balance = ctx.accounts.authority_collateral.amount;
+
+    // 1 - calculate the value of collectable interests and fees (in USDC unit)
+    let collectable_profits_value = ctx.accounts.calculate_collectable_profits_value()?;
+
+    // 2 - calculate the amount of LP token to withdraw to collect all interests and fees
+    // This LP amount is subjected to precision loss (we handle this precision loss later)
+    let lp_token_amount_to_match_collectable_profits_value = ctx
+        .accounts
+        .calculate_lp_token_amount_to_match_collectable_profits_value(collectable_profits_value)?;
+
+    let possible_lp_token_precision_loss_collateral_value =
+        mercurial_utils::calculate_possible_lp_token_precision_loss_collateral_value(
+            &ctx.accounts.mercurial_vault,
+            ctx.accounts.mercurial_vault_lp_mint.supply,
+        )?;
+
+    // 3 - withdraw collateral from mercurial vault for LP tokens
+    mercurial_vault::cpi::withdraw(
+        ctx.accounts
+            .into_withdraw_collateral_from_mercurial_vault_context()
+            .with_signer(depository_signer_seed),
+        lp_token_amount_to_match_collectable_profits_value,
+        // Do not check slippage here
+        0,
+    )?;
+
+    // 4 - Reload accounts impacted by the withdraw (We need updated numbers for further calculation)
+    ctx.accounts.depository_lp_token_vault.reload()?;
+    ctx.accounts.authority_collateral.reload()?;
+
+    // 5 - Check that a positive amount of collateral have been redeemed
+    let after_collateral_balance = ctx.accounts.authority_collateral.amount;
+
+    let collateral_balance_change = after_collateral_balance
+        .checked_sub(before_collateral_balance)
+        .ok_or_else(|| error!(UxdError::MathError))?;
+
+    require!(
```

Comment:
Note to myself, we shouldn't throw this error if there are no profit to collect. Maybe a NoProfitToCollect error instead or no error at all and just leave the function